apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx

```diff
@@ -0,0 +1,254 @@
+import { ChangeEvent, FocusEvent, KeyboardEvent, useRef } from "react";
+import {
+  Box,
+  Text,
+  Grid,
+  IconButton,
+  TextField,
+  Tooltip,
+  IconButtonWithMenu,
+  numericScrubControl,
+} from "@webstudio-is/design-system";
+import { getFinalValue } from "../../shared/get-final-value";
+import { useIsFromCurrentBreakpoint } from "../../shared/use-is-from-current-breakpoint";
+import { ControlProps } from "../../style-sections";
+import { iconConfigs, StyleConfig } from "../../shared/configs";
+import { type StyleValue, type Unit, units } from "@webstudio-is/react-sdk";
+import { ChevronDownIcon } from "@webstudio-is/icons";
+import { PropertyName } from "../../shared/property-name";
+
+const commonUnitsSubset = units
+  .slice(0)
+  .sort((v) =>
+    ["%", "px", "rem", "em", "ch", "vh", "vw", "hv", "vmin", "vmax"].includes(v)
+      ? -1
+      : 1
+  );
+
+const gridTemplateStyle = {
+  gridTemplateColumns: "$sizes$6 auto $sizes$6",
+  gridTemplateRows: "repeat(1, 1fr)",
+};
+
+const gridLeftStyle = {
+  "& ~ input": {
```

Comment:
I have been just discussing the kind of selectors in https://github.com/webstudio-is/webstudio-designer/pull/251

The problem is that you are overriding using cascade from parent into the child, where the child is not "your" component. It's a component that knows nothing about what you are doing from here.

This has the potential for specificity issues and accidental visual regressions.

Instead, we could use some explicit way of passing down CSS values to the child: using a prop or using a CSS variable.

Please review the other PR, there are still some open options there and then we can apply the solution here.